File: atividade-4/053_deletee/053_deletee.hs
module Main where

-- Remover a primera ocorrencia de algum elemento no vetor

deletee :: Int -> [Int] -> [Int]
deletee n [] = []
deletee n (x:xs)
            | x == n = xs
            | otherwise = x:deletee n xs

main = do
  a <- readLn :: IO Int
  b <- readLn :: IO [Int]
  print $ deletee a b

File: atividade-4/053_deletee/053_deletee.js
// Remover a primera ocorrencia de algum elemento no vetor

const number = 3;
const list = [1, 3, 5, 7, 9, 3];

const deletee = (n, xs) => {
  const pos = xs.indexOf(n);

  if (pos !== -1) xs.splice(pos, 1);

  return xs;
};

console.log(deletee(number, list));
